ceph/rados/core_workflows.py

```diff
@@ -2195,3 +2195,135 @@ def create_inconsistent_object(self, pool_name, object_name):
+                try:
+                    health, _ = self.node.shell(args=["ceph health detail"])
+                    log.info(f"Health warning: \n {health}")
+                    assert "daemons have recently crashed" in health
+                    assert f"{daemon}.{id} crashed" in health
+                    return True
+                except AssertionError:
+                    log.error(
+                        "Crash error is yet to appear in ceph health. Sleeping for 30 secs"
+                    )
+                    time.sleep(30)
+                    if datetime.datetime.now() >= end_time:
+                        log.error("daemon crash warning not found within timeout")
+                        return False
+
+        # check if input daemon exists on the cluster
+        status, desc = self.get_daemon_status(daemon_type=daemon, daemon_id=id)
+        if int(status) != 1 or "running" not in desc:
+            log.error(
+                f"Input daemon {daemon}.{id} either does not exist or "
+                f"currently not running."
+            )
+            return False
+
+        # fetch daemon host
+        daemon_host = self.fetch_host_node(daemon_type=daemon, daemon_id=id)
+
+        # ensure crash module is enabled
+        self.node.shell(["ceph mgr module enable crash"])
+
+        # capture initial number of crash reports in the /var/lib/ceph/crash directory
+        _cmd = "cephadm shell -- ls /var/lib/ceph/crash/ | wc -l"
+        init_crash_reports, err = daemon_host.exec_command(cmd=_cmd, sudo=True)
+        # capture initial number of crash list from crash ls-new
+        init_crash_ls = len(self.run_ceph_command("ceph crash ls-new"))
+
+        # generate crash using ceph daemon command
+        asok_cmd = (
+            f"cephadm shell -- ceph daemon /var/run/ceph/ceph-{daemon}.{id}.asok assert"
+        )
+        try:
+            out, err = daemon_host.exec_command(cmd=asok_cmd, sudo=True)
+        except Exception as err_exec:
+            if "exception" not in str(err_exec):
+                log.error(f"STDERR: {err_exec}")
+                log.error("Could not crash the daemon with ceph daemon utility")
+                return False
+
+        crash_report_post, err = daemon_host.exec_command(cmd=_cmd, sudo=True)
+        if not crash_report_post > init_crash_reports:
+            log.error("New crash report directory not generated in /var/lib/ceph/crash")
+            return False
+        log.info("New crash report generated successfully")
+
+        # get the crash report directory
+        _cmd = "cephadm shell -- ls -rt /var/lib/ceph/crash/ | tail -1"
+        dir, err = daemon_host.exec_command(cmd=_cmd, sudo=True)
+        dir = str(dir).strip()
+
+        # copy the crash directory to tmp
+        fsid = self.run_ceph_command(cmd="ceph fsid")["fsid"]
+        _cmd = f"cp -r /var/lib/ceph/{fsid}/crash/'{dir}' /tmp/"
+        daemon_host.exec_command(cmd=_cmd, sudo=True)
```

Comment:
Do we want to do ls and make sure the dir is copied?

Reply:
It implicitly gets verified when we are doing a `ceph crash post -i`, if the directory is not copied correctly, crash warnings won't show up
I can add this check, but I feel it would be redundant
let me know if the check is necessary, I can still accommodate it

Reply:
can be dropped then.